### PR TITLE
Modifies LightningFromCapePrecip plugin to accept 3h precipitation-rate-max data

### PR DESCRIPTION
**Pull request by MoseleyS**
Addresses #1518

We need this so we can generate data for days 6-8 when only 3h data are available.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Review by gavinevans — CHANGES_REQUESTED** [review]
Thanks @MoseleyS 👍 

I've added a few minor comments.

File: improver/lightning.py [hunk 1]
```diff
@@ -74,7 +74,8 @@ def _get_inputs(cubes: CubeList) -> Tuple[Cube, Cube]:
         """
         Separates CAPE and precipitation rate cubes and checks that the following
         match: forecast_reference_time, spatial coords, time-bound interval and
-        that CAPE time is exactly one hour older than precipitation rate time.
+        that CAPE time is at the lower bound of precipitation rate time.
```
[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver/lightning.py:77]: Fixed. Final tweak from #1550 hadn't been committed and merged - it has now.

File: improver_tests/lightning/test_LightningFromCapePrecip.py [hunk 2]
```diff
@@ -142,21 +155,18 @@ def test_with_model_attribute(cape_cube, precip_cube, expected_cube):
...
+    return r"CAPE cube time .* should be valid at the precipitation_rate_max cube lower bound .*"
 
 
 def break_time_bound(cape_cube, precip_cube):
     """Modifies lower bound on precip_cube time coord to be incremented by 1 second and
     returns the error message this will trigger"""
     bounds = precip_cube.coord("time").bounds
-    precip_cube.coord("time").bounds = (bounds[0][0] + 1, bounds[0][1])
-    return r"Precipitation_rate_max cube time window must be one hour, not .*"
+    precip_cube.coord("time").bounds = (bounds[0][0], bounds[0][1] + 1)
```
[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver_tests/lightning/test_LightningFromCapePrecip.py:168]: Done

File: improver_tests/lightning/test_LightningFromCapePrecip.py [hunk 3]
```diff
@@ -129,6 +129,19 @@ def test_basic(cape_cube, precip_cube, expected_cube):
     assert np.allclose(result.data, expected_cube.data)
 
 
+def test_3d_cubes(cape_cube, precip_cube, expected_cube):
+    """Run the plugin again with 3h cubes"""
+    cape_cube.coord("time").points = cape_cube.coord("time").points - 2 * 3600
+    bounds = precip_cube.coord("time").bounds
```
[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver_tests/lightning/test_LightningFromCapePrecip.py:135]: Good point. Done

File: improver_tests/lightning/test_LightningFromCapePrecip.py [hunk 4]
```diff
@@ -129,6 +129,19 @@ def test_basic(cape_cube, precip_cube, expected_cube):
     assert np.allclose(result.data, expected_cube.data)
 
 
+def test_3d_cubes(cape_cube, precip_cube, expected_cube):
```
[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver_tests/lightning/test_LightningFromCapePrecip.py:132]: :-O   Fixed

File: improver_tests/lightning/test_LightningFromCapePrecip.py [hunk 5]
```diff
@@ -142,6 +143,7 @@ def test_3d_cubes(cape_cube, precip_cube, expected_cube):
     assert np.allclose(result.data, expected_cube.data)
 
 
+
```
Comment (gavinevans) [inline improver_tests/lightning/test_LightningFromCapePrecip.py:146]: Black is currently failing and it's probably because of this blank line.

**Review by gavinevans — CHANGES_REQUESTED** [review]

**Review by gavinevans — APPROVED** [review]

**Review by bayliffe — APPROVED** [review]
This looks fine, and all the tests pass. I've nothing to add.

**Pull request merged**